nltk/metrics/distance.py

```diff
@@ -182,18 +186,62 @@ def custom_distance(file):
             labelA, labelB, dist = l.strip().split("\t")
             labelA = frozenset([labelA])
             labelB = frozenset([labelB])
-            data[frozenset([labelA,labelB])] = float(dist)
-    return lambda x,y:data[frozenset([x,y])]
+            data[frozenset([labelA, labelB])] = float(dist)
+    return lambda x, y: data[frozenset([x, y])]
+
+
+def jaro_winkler_distance(s1, s2, p=0.1):
+    """ Details about this implementation can be found at
+    https://en.wikipedia.org/wiki/Jaro-Winkler_distance
+dist_bound = type int, upper bound of distance for being a matched character
+mtch = type int, no.of matched characters in s1 and s2
+t_cnt= half of the number of transpositions between s1 and s2
+j_sim = type float ,jaro similarity between s1 and s2
+p = type float , weight parameter with a standard value of 0.1
+l_cnt = type int, common prefix at the start of the string (max val = 4)
+jw_sim= type float, jaro winkler similarity between s1 and s2
+jw_dist = type float, it is equal to 1 - jw_sim """
```

Comment:
The docstring has to be a lot clearer. 


- **What's s1 and s2?** I assume it's taking strings as the distance but actually the other distance metrics in `nltk.metrics.distance` accepts any sequence type. Jaro-Winkler seems to be generic enough to cover any sequence.

Reply:
Also variable names in this function needs to be clearer. 

Since the function scopes within itself, you can afford to have more explicit variable names, e.g.

 - `mtch` -> `matches`
 -  `t_cnt` -> `transposition_counts`

Reply:
s1 and  s2 are the 2 strings between which we wanna calculate the JW distance. I am going to include it in the docstring. Thanks for the point out